### PR TITLE
Add missing Ubuntu 20.04 and 24.04 helix images

### PR DESCRIPTION
**Pull request by richlander**
I added a complete architecture compliment for 20.04 and 24.04. We can add missing ones to 22.04 when 20.04 goes EOL. No pressing reason now, AFAIK.

### CODE REVIEW

**Comment by richlander** [comment]
I absolutely abused CI on this PR. It is a bit shameful. I'll try to avoid that in future.

I pulled the Arm32 Dockerfile since it is running into the Y2038/time_t issue with msquic. We can take that up as a focused PR.

**Comment by riarenas** [comment]
/azp run dotnet-buildtools-prereqs-docker-almalinux,dotnet-buildtools-prereqs-docker-alpine,dotnet-buildtools-prereqs-docker-azurelinux,dotnet-buildtools-prereqs-docker-cbl-mariner,dotnet-buildtools-prereqs-docker-centos,dotnet-buildtools-prereqs-docker-debian,dotnet-buildtools-prereqs-docker-eng,dotnet-buildtools-prereqs-docker-fedora,dotnet-buildtools-prereqs-docker-nanoserver,dotnet-buildtools-prereqs-docker-nanoserver,dotnet-buildtools-prereqs-docker-opensuse,dotnet-buildtools-prereqs-docker-raspbian,dotnet-buildtools-prereqs-docker-ubuntu,dotnet-buildtools-prereqs-docker-windowsservercore

**Comment by richlander** [comment]
@mthalman -- looking for an approval. thanks.

File: src/ubuntu/24.04/helix/amd64/Dockerfile
Comment (mthalman) [inline src/ubuntu/24.04/helix/amd64/Dockerfile:1]: I see some non-architecture-specific diffs between this Dockerfile and the Arm64 one.

image

Those should be resolved.

What are your thoughts on making a single Dockerfile used for both architectures? That would help with maintenance. That would just need some minor adjustments to dynamically derive the architecture-specific DEB filenames. Same could be done for 20.04.

**Comment by richlander** [comment]
@mthalman -- good to go?

File: src/ubuntu/manifest.json [hunk 1]
```diff
@@ -334,7 +352,7 @@
           "platforms": [
             {
               "architecture": "amd64",
-              "dockerfile": "src/ubuntu/22.04/helix/amd64",
+              "dockerfile": "src/ubuntu/22.04/helix/any",
```
Comment (mthalman) [inline src/ubuntu/manifest.json:355]: While I'm indifferent to the use of the `any` directory pattern in general, it doesn't align with the existing pattern in this repo for multi-arch Dockerfiles. For example, https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src/ubuntu/20.04/Dockerfile is a multi-arch Dockerfile that is placed at the "root". In the context of this PR, that would mean placing the Dockerfile at `src/ubuntu/22.04/helix`.

Comment (mthalman) [reply src/ubuntu/manifest.json:355]: Yes please

Comment (richlander) [reply src/ubuntu/manifest.json]: Done

**Review by mthalman — APPROVED** [review]

**Pull request merged**